src/App.jsx

```diff
@@ -1,21 +1,32 @@
-import { useState } from "react";
+import { useState, useEffect, Fragment } from "react";
 import "./App.css";
 import TodoList from "./TodoList";
 import AddToDoForm from "./AddTodoForm";
 
-function App() {
-  const [todoList, setTodoList] = useState([]);
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
```

Comment:
It's preferable to pass an initializer function as argument to `useState` when the initial value does expensive operations. Reading from local storage and parsing a string to JSON can be both considered expensive, so running them only once instead of on every render will improve performance.

```suggestion
    () => JSON.parse(localStorage.getItem("savedTodoList")) || []
```

The reason is that every time React needs to render the page, it will run the component function. Even though the initial state is already set, the instructions to parse and read from the storage are there and will be executed again. When you pass a function to `useState`, React will make sure of running it only once.